x/concentrated-liquidity/swapstrategy/zero_for_one.go

```diff
@@ -77,6 +81,21 @@ func (s zeroForOneStrategy) ComputeSwapWithinBucketOutGivenIn(sqrtPriceCurrent,
 		sqrtPriceNext = math.GetNextSqrtPriceFromAmount0InRoundingUp(sqrtPriceCurrent, liquidityBigDec, amountZeroInRemainingLessSpreadReward)
 	}
 
+	// The code snippet below deals with an edge case stemming from rounding.
+	// If the computed sqrtPriceNext is 1 BigDec ULP away from the sqrtPriceTarget AND
```

Comment:
Is it 1 ULP or 1 smallestBigDec? Or are these synonymous?

Reply:
Synonymous in this context. BigDec and Dec ULPs are different.

This is what this stands for: https://en.wikipedia.org/wiki/Unit_in_the_last_place